src/qt/dc.cpp

```diff
@@ -115,14 +115,46 @@ bool wxQtDCImpl::CanGetTextExtent() const
 
 void wxQtDCImpl::DoGetSize(int *width, int *height) const
 {
-    if (width)  *width  = m_qtPainter->device()->width();
-    if (height) *height = m_qtPainter->device()->height();
+    QPaintDevice *pDevice = m_qtPainter->device();
+
+    int deviceWidth;
+    int deviceHeight;
+
+    if (pDevice)
+    {
+        deviceWidth = pDevice->width();
+        deviceHeight = pDevice->height();
+    }
+    else
+    {
+        deviceWidth = 0;
+        deviceHeight = 0;
+
+    }
+    if (width) *width  = deviceWidth;
```

Comment:
As long as you're changing these lines, they could be brought in line with wxW coding style (which could include `{` `}` even for single line blocks).
```suggestion
    if ( width )
        *width = deviceWidth;
```